src/main/java/components/Parser.java

```diff
@@ -0,0 +1,229 @@
+package components;
+
+import command.*;
+import exception.*;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.ToDo;
+import task.Task.TaskType;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a handler that parses user input.
+ */
+public class Parser {
+    /**
+     * Returns a Command object that can execute the user input.
+     *
+     * @param input Input by user.
+     * @return Command to be executed based on input.
+     * @throws NyabotException If exception arises during parsing. Examples include
+     * NyabotNoSuchCommandException when there is no relevant command.
+     */
+    public static Command parse(String input) throws NyabotException {
+        String firstWord = input.split(" ")[0];
+        switch(firstWord) {
+        case "bye":
+            return new ExitCommand();
+        case "list":
+            return new ListCommand();
+        case "mark": case"unmark":
+            int number = handleTaskWithTaskNumber(input, firstWord);
+            return new MarkCommand(number, firstWord.equals("mark"));
+        case "find":
+            String searchTerm = parseTaskWithWord(input, "find");
+            return new FindCommand(searchTerm);
+        case "deadline": case "todo": case "event":
+            task.Task.TaskType taskType =  firstWord.equals("todo")
+                    ? task.Task.TaskType.TODO
+                    : firstWord.equals("deadline")
+                    ? task.Task.TaskType.DEADLINE
+                    : task.Task.TaskType.EVENT;
+            return parseNewTask(input, taskType);
+        case "delete":
+            int deleteNumber = handleTaskWithTaskNumber(input, "delete");
+            return new DeleteCommand(deleteNumber);
+        case "save":
+            return new SaveCommand();
+        default:
+            throw new NyabotNoSuchCommandException("Nya, there is no such command!");
+        }
+    }
+
+    /**
+     * Returns string of words the user wrote after they wrote the command name.
+     *
+     * @param input Input by user that has a variable series of words after the command name.
+     * @param command Command name as listed in the input.
+     * @return Words after the command name in the input.
+     * @throws NyabotMissingArgumentException If no content is written after the command
+     * in the input.
+     */
+    public static String parseTaskWithWord(String input, String command) throws NyabotMissingArgumentException {
```

Comment:
Should this method be public? I noticed that it was only used within the Parser class itself.